internal/providers/terraform/aws/eks_node_group.go

```diff
@@ -0,0 +1,138 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+func GetNewEKSNodeGroupItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "aws_eks_node_group",
+		RFunc: NewEKSNodeGroup,
+		ReferenceAttributes: []string{
+			"launch_template.0.id",
+		},
+	}
+}
+
+func NewEKSNodeGroup(d *schema.ResourceData, u *schema.ResourceData) *schema.Resource {
+	region := d.Get("region").String()
+	scalingConfig := d.Get("scaling_config").Array()[0]
+	desiredSize := scalingConfig.Get("desired_size").Int()
+	instanceType := "t3.medium"
+	if d.Get("instance_types").Exists() {
+		// Only a single type is expected https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_node_group#instance_types
+		instanceType = d.Get("instance_types").Array()[0].String()
+	}
+
+	launchTemplateRef := d.References("launch_template.0.id")
+
+	costComponents := make([]*schema.CostComponent, 0)
+	subResources := make([]*schema.Resource, 0)
+
+	if len(launchTemplateRef) > 0 {
+		onDemandCount := decimal.NewFromInt(desiredSize)
+		spotCount := decimal.Zero
+		if launchTemplateRef[0].Get("instance_market_options.0.market_type").String() == "spot" {
+			onDemandCount = decimal.Zero
+			spotCount = decimal.NewFromInt(desiredSize)
+		}
+		lt := newLaunchTemplate(launchTemplateRef[0].Address, launchTemplateRef[0], region, onDemandCount, spotCount)
+
+		// AutoscalingGroup should show as not supported LaunchTemplate is not supported
+		if lt == nil {
+			return nil
+		}
+		subResources = append(subResources, lt)
+	} else {
+		costComponents = append(costComponents, eksComputeCostComponent(d, region, desiredSize, instanceType))
+		eksCPUCreditsCostComponent := eksCPUCreditsCostComponent(d, region, desiredSize, instanceType)
+		if eksCPUCreditsCostComponent != nil {
+			costComponents = append(costComponents, eksCPUCreditsCostComponent)
+		}
+		costComponents = append(costComponents, newEksRootBlockDevice(d))
+	}
+
+	return &schema.Resource{
+		Name:           d.Address,
+		CostComponents: costComponents,
+		SubResources:   subResources,
+	}
+}
+
+func eksComputeCostComponent(d *schema.ResourceData, region string, desiredSize int64, instanceType string) *schema.CostComponent {
```

Comment:
I think we can call the `computeCostComponent` method in `instance.go` and then call `multiplyQuantities` (from `autoscaling_group.go`) with desiredSize. That means we can delete this method.
We could also move `multiplyQuantities` to `util.go`

Reply:
`computeCostComponent` is coupled to hows the JSON schema is structure. E.g. instance.go uses `insatnce_type` while node group uses `instance_types` (even though there's only 1 element allowed)

Reply:
I think if we want to DRY this, we need to refactor both instance and node group into shared cost components that define the arguments they need. Probably have them live away from either of the resource files too. Maybe `internal/providers/terraform/aws/shared_cost_components/` folder for example?

Reply:
In either case, I suggest performing the DRYing work as a separate PR, not to delay this feature.

Reply:
agreed!